runtime/lua/vim/shared.lua

```diff
@@ -228,7 +219,9 @@ function vim.tbl_extend(behavior, ...)
     vim.validate{["after the second argument"] = {tbl,'t'}}
     if tbl then
       for k, v in pairs(tbl) do
-        if behavior ~= 'force' and ret[k] ~= nil then
+        if type(v) == 'table' and deep_extend and not vim.tbl_islist(v) then
```

Comment:
Why `and not vim.tbl_islist(v)`?

I would expect deep extend to also copy table elements?

```lua
points = { { x = 1, y = 1}, { x = 3, y = 5} }
```

Reply:
Oh, I see -- it doesn't work for non-map-like tables. Why is that?

Reply:
Actually, nvm, this doesn't apply.